applications/directory/src/components/CreateAccount.js

```diff
@@ -63,31 +130,66 @@ function CreateAccount() {
             }}
             sx={{ minWidth: ['100%', '100%', '325px'] }}
             required
+            onChange={handleChange}
+            onBlur={async (e) => {
+              if (values.email) {
+                const ifAccountExists = await ifEmailExists(values.email);
+                // if email is valid (there is a profile with this email in DB) set error message to empty string or populate with proper error message
+                if (ifAccountExists.error) {
+                  setIfValidEmail('Server error. Fail to verify email');
+                } else if (!ifAccountExists.profile) {
+                  setIfValidEmail(
+                    'No Profile found associated with provided email'
+                  );
+                } else if (ifAccountExists.account) {
+                  setIfValidEmail('Account with given email already exists');
+                } else {
+                  setIfValidEmail('');
+                }
+              }
+              handleBlur(e);
+            }}
+            value={values.email}
+            error={errors.email && touched.email}
+            helperText={touched.email ? errors.email : ''}
           />
         </Box>
         <Box sx={{ mb: 2 }}>
           <TextField
+            required
+            type="password"
             id="outlined-password-input"
             label="Password"
-            type="password"
+            name="password"
             autoComplete="current-password"
+            onChange={handleChange}
+            onBlur={handleBlur}
+            value={values.password}
+            error={errors.password && touched.password}
+            helperText={touched.password ? errors.password : ''}
```

Comment:
It looks like `helperText` is being used to display an error. I wouldn't do this, as `helperText` should be used to display helpful messages, not error messages. I suggest using `error={errors.password}` to display the error.

I think using `Field` from Formik could greatly simply things for you. [Here](https://smithfield.sfdbrands.com/en-us/contact/general-inquiry) is an example form we built at my last job, and here is one of the fields in that form:

```jsx
<Field
  as={TextField}
  id="email"
  name="email"
  label="Email"
  placeholder="Enter email"
  required
  error={errors.email}
  touched={touched.email}
/>
```

Notice a few things:
- `as={TextField}`
- `error={errors.email}`
- `touched={touched.email}`
- No `onChange`, `onBlur`, or `value` -- these get added automatically when using `Field` + `as`

Reply:
If using TextField from MUI it does not display the error without using helperText prop. How about - i keep it in mind, try it and then If it works I will change it in all form fields it is used like it?

Reply:
Oh, [I see](https://mui.com/material-ui/react-text-field/#validation). Okay, I understand. It seems they promote use of `helperText` to display errors, so let's continue doing that.